zarr/core.py

```diff
@@ -1498,7 +1498,11 @@ def _set_selection(self, indexer, value, fields=None):
         sel_shape = indexer.shape
 
         # check value shape
-        if is_scalar(value, self._dtype):
+        if sel_shape == ():
+            # setting a single item
+            pass
+        elif is_scalar(value, self._dtype):
+            # setting a scalar value
```

Comment:
Just to be clear, this is more of a general change to how selection works and not something that is specific to `object` type data (though it likely will be used for that case), correct?

Reply:
This is necessary to allow storing a list as an item in an object array. I.e., 

```
z = zarr.empty(10, dtype=object)
z[0] = ['foo', 'bar', 'baz']
```

...needs this change to work.

Reply:
Sorry. Meant to put that comment near the `()` piece above. The scalar part is clear. :)

Reply:
No worries. Yes it is a general thing, although it is really an optimisation for the case of setting a single item, and shouldn't change behaviour for anything other than allowing more flexibility in what kinds of items can be set in object arrays.